src/dmd/lexer.d

```diff
@@ -1943,8 +1950,13 @@ class Lexer
             case '5':
             case '6':
             case '7':
+                base = 8;
+                break;
+
             case '8':
             case '9':
+                if (Ccompile)
+                    error("octal constants consist of digits `0` thru `7`, not `%c`", c);
```

Comment:
thru => through

Reply:
thru is a common abbrevation

Reply:
Wiktionary describes it as "(Canada, US, informal) Rare spelling of through." or "(Britain, rare, regional or dialectal) Eye dialect spelling of through.". It seems weird to me to use colloquial language here just to save 3 characters.

Reply:
> thru is a common abbrevation

Common in the same way as whom is common.  It's regularly used in one phrase (fall-thru), and seldom used correctly outside of that context.

Reply:
How about ``error("invalid digit `%c` in octal constant", c)`` ?

Reply:
My plan is build the bike shed before debating the color of it :-)

Reply:
@WalterBright we the existing error elsewhere in the DMD sources is:
```
error("octal digit expected, not `%c`", c);
```
Why can't we just recycle the same?

Reply:
Grepping for that message reveals no hits.

Reply:
https://github.com/dlang/dmd/blob/7555bd962776847d25f3849aca852c72047ab526/src/dmd/lexer.d#L2054

Reply:
ok